core/src/Node.ts

```diff
@@ -412,42 +412,83 @@ export class LoaderNode extends Node implements LoaderNodeProps {
   }
 }
 
-export interface ElementNodeProps extends NodeProps {
-
+export type ElementNodeProps = NodeProps & {
+  /*
+   * References a Category Node defined by user
+   */
+  category?: ElementNode,
+} & ({
   /*
    * Allows multiple EC Elements to be populated by a single ElementNode
    */
-  dmo: ElementDMO;
+  dmo: ElementDMO,
+
+  /*
+   * If an element has a model, it must not have a parent.
+   *
+   * No types are assignable to never. The DMO prevents us from just leaving the parent and model
+   * properties off of the object type because it narrows the union.
+   *
+   * @see https://stackoverflow.com/a/44425486
```

Comment:
@admccarthy1 @akshay-madhukar  let's help Jackson think about this design.  Please checkout the stackoverflow post he referenced.  Noticed how followed the pattern in the article to achieve this exclusive or behavior.  i.e. ElementNodeProps must have either a parent (element) or a model but not both.  The old code required a model, so whatever refactoring is done should support that.  I suggested an interface such as IParental and have ModelNode and ElementNode implement IParental.  Then, rather than requiring this XOR behavior, we could keep the model property but change the type to IParental.  That way doesn't require the new, parent property nor the convoluted XOR pattern.  What are your thoughts here?